embassy-net-driver/Cargo.toml

```diff
@@ -21,4 +21,5 @@ target = "thumbv7em-none-eabi"
 features = ["defmt"]
 
 [dependencies]
-defmt = { version = "0.3", optional = true }
\ No newline at end of file
+defmt = { version = "0.3", optional = true }
+smoltcp = { version = "0.10", default-features = false }
```

Comment:
`embassy-net-driver` should not depend on smoltcp. This is so smoltcp can be upgraded without requiring every driver crate to update. Can you copypaste the `HardwareAddress` enum instead?

Reply:
> Can you copypaste the `HardwareAddress` enum instead?

IMO that loses part of the meaning of using `HardwareAddress`, since the whole purpose is to be able to use it as-is with `smoltcp`. It would be cool if we could just [depend on smoltcp_wire](https://github.com/smoltcp-rs/smoltcp/pull/812) instead, I suppose, although even that doesn't alleviate the concern.

I get the concern however, it felt pretty awkward to put smoltcp as a dependency.

Another option that I didn't mention in my OP, is to fix the hardware address type at compile time through a generic.

If after my plea here you still rather have the definition copied, I'll happily do that!

Reply:
I slept on it, and I think I agree. So I'll rewrite it and fix CI. If we're happy with the result, I'll squash the results such that Cargo.toml remains unchanged.